src/components/album/AlbumChatBox.js

```diff
@@ -78,6 +78,23 @@ const AlbumChatBox = ({ spotifyUrl }) => {
     };
   }, [albumId, socket, loggedInUserId]);
 
+  useEffect(() => {
+    const fetchRecentMessages = async () => {
+      if (albumId && !socket) {
+        const response = await fetch(
+          `${process.env.REACT_APP_API_BASE_PATH}/chat/${albumId}`
+        );
+        const data = await response.json();
+        const messagesWithKeys = data.messages.map((message) => ({
+          ...message,
+          user: { name: message.user },
```

Comment:
Looks good; I'm guessing we may need to change this once you've updated the backend to send the full user object for this endpoint?

Reply:
Do I need to import the User model in handleAlbumChat.js to fetch all the user data on the backend? Or should I add the name directly to the ChatMessage model? Although it doesn't make sense to include it in ChatMessage.js when we already have the data in the User model. Now i don't have a name 
<img width="805" alt="2" src="https://github.com/Code-the-Dream-School/dd-prac-team4-front/assets/95896159/ea8ae289-89aa-482a-98b0-845d4d51ec45">

Reply:
Sorry for my recent "sleepy commits"😅

Reply:
Since `ChatMessage` schema tells mongoose that `user` field is a "ref" (reference) to the `User` model, we can use the `populate` function. It will be the very same as how you've done it in the chat socket to populate the user name here: https://github.com/Code-the-Dream-School/dd-prac-team4-back/blob/7d57b27f679d6480f43d147f86dcbd305e8b9347/src/live/handleAlbumChat.js#L12

Only you'd be doing it in the `chatRoutes` file where you fetch the chat messages.